video_xblock/static/js/player_state.js

```diff
@@ -30,6 +30,26 @@ var player_state = {
 };
 
 var xblockUsageId = window.location.hash.slice(1);
+var transcripts = {
+  {% for transcript in player_state.transcripts %}
+   '{{transcript.lang}}': {
+          'label': '{{transcript.label}}',
+          'url': '{{transcript.url}}',
+   },
+  {% endfor %}
+};
+
+/** Get transcript url for current caption language and switch off transcripts and captions state if doesn`t have one */
+var getDownloadTranscriptUrl = function (player){
+  var downloadTranscriptUrl;
+  if (transcripts[player.captionsLanguage]){
+    downloadTranscriptUrl = transcripts[player.captionsLanguage].url;
+  } else {
+    downloadTranscriptUrl = '#';
+    player.captionsEnabled = player.transcriptsEnabled = false;
```

Comment:
Remove this side-effect out of this function. It's better to explicitly do this on line 75